### PR TITLE
Segprint: print related tiles

### PR DESCRIPTION
**Pull request by mcmasterg**
Fixes https://github.com/SymbiFlow/prjxray/issues/218

### CODE REVIEW

File: utils/segprint.py [hunk 1]
```diff
@@ -20,7 +22,6 @@ class NoDB(Exception):
 # int and sites are loaded together so that bit coverage can be checked together
 # however, as currently written, each segment is essentially printed twice
 def process_db(db, tile_type, process, verbose):
-    print(db.get_tile_types())
     ttdb = db.get_tile_type(tile_type)
 
     fns = [ttdb.tile_dbs.segbits, ttdb.tile_dbs.ppips]
```
Comment (litghost) [inline utils/segprint.py]: Could use TileSegbits via db.get_tile_segbits? 
https://github.com/SymbiFlow/prjxray/blob/master/prjxray/db.py#L138

File: prjxray/db.py [hunk 2]
```diff
@@ -25,14 +25,15 @@ def get_available_databases(prjxray_root):
 
 
 class Database(object):
-    def __init__(self, db_root):
```
Comment (litghost) [inline prjxray/db.py]: The merge resolution is backwards.  Tools should always specify the db_root, rather than relying on implicit env vars.

Comment (mcmasterg) [reply prjxray/db.py]: I disagree. I want convenient tools

Comment (mcmasterg) [reply prjxray/db.py]: The alternate solution would be a util class to make the DB. Does that work for you?

Comment (litghost) [reply prjxray/db.py]: The tools can default to the env vars, but need to work in the absence of those env vars.  Env vars need to be kept in the roots of tools as much as possible, especially for tools that will be used extensively outside of database generation.

File: utils/segprint.py [hunk 3]
```diff
@@ -222,18 +398,25 @@ def tile_segnames(tiles):
...
         segnames,
         omit_empty_segs=False,
         flag_unknown_bits=False,
         flag_decode_emit=False,
         flag_decode_omit=False,
         verbose=False):
-    tiles, segments = mk_grid(db_root)
-    db = prjxraydb.Database(db_root)
-
+    db = prjxraydb.Database()
```
Comment (litghost) [inline utils/segprint.py]: Both mk_grid and Database rely on a common db_root, this should be made explicit.

File: prjxray/db.py (hunk 2 above)
Comment (litghost) [reply prjxray/db.py]: I'm fine with a util function to make the DB, but in this case the tool needs to be able to take the db_root from the command line, so the util function is redundant.

Comment (mcmasterg) [reply prjxray/db.py]: Ah I see. I misunderstood what the merge conflict was. Let me re-review

File: utils/segprint.py [hunk 4]
```diff
@@ -231,9 +415,9 @@ def run(
         flag_decode_emit=False,
         flag_decode_omit=False,
         verbose=False):
-    tiles, segments = mk_grid(db_root)
     db = prjxraydb.Database(db_root)
-
+    tiles = load_tiles()
```
Comment (litghost) [inline utils/segprint.py]: Have load_tiles take db_root, and I think we are good.

**Comment by mcmasterg** [comment]
@litghost sorry, misunderstood what the issue was. I've more carefully reviewed the merge issues and submitted a new patch. Please review and let me know if this works for you

File: utils/segprint.py (hunk 1 above)
Comment (litghost) [reply utils/segprint.py]: Sounds good to me.

**Comment by litghost** [comment]
@mcmasterg No problem, left a new comment, and I think we are good to go.

**Pull request merged**